File: MvvmSample/app/src/main/java/com/abben/mvvmsample/vo/TypeMovies.java
package com.abben.mvvmsample.vo;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Created by abben on 2017/10/27.
 */
public class TypeMovies {

    public final static String API_TYPE_ALL_MOVIES = "Movies";
    public final static String API_TYPE_CHINESE_MOVIES = "ChineseMovies";
    public final static String API_TYPE_EURAMERICAN_MOVIES = "EuramericanMovies";
    public final static String API_TYPE_JANPAN_AND_KOREA_MOVIES = "JanpanAndKoreaMovies";

    public final static String TYPE_ALL_MOVIES_ZH = "所有电影";
    public final static String TYPE_CHINESE_MOVIES_ZH = "国产电影";
    public final static String TYPE_EURAMERICAN_MOVIES_ZH = "欧美电影";
    public final static String TYPE_JANPAN_AND_KOREA_MOVIES_ZH = "日韩电影";

    @StringDef({TYPE_ALL_MOVIES_ZH, TYPE_CHINESE_MOVIES_ZH, TYPE_EURAMERICAN_MOVIES_ZH, TYPE_JANPAN_AND_KOREA_MOVIES_ZH})
    @Retention(RetentionPolicy.SOURCE)
    public @interface TypeMoviesAnnotation {
    }
}

File: MvvmSampleJetpack/app/src/main/java/com/abben/mvvmsamplejetpack/vo/TypeMovies.kt
package com.abben.mvvmsamplejetpack.vo

import androidx.annotation.StringDef

/**
 *  @author abben
 *  on 2019/1/17
 */
const val TYPE_ALL_MOVIES_ZH: String = "所有电影"
const val TYPE_CHINESE_MOVIES_ZH: String = "国产电影"
const val TYPE_EURAMERICAN_MOVIES_ZH: String = "欧美电影"
const val TYPE_JANPAN_AND_KOREA_MOVIES_ZH: String = "日韩电影"


@StringDef(
    TYPE_ALL_MOVIES_ZH,
    TYPE_CHINESE_MOVIES_ZH,
    TYPE_EURAMERICAN_MOVIES_ZH,
    TYPE_JANPAN_AND_KOREA_MOVIES_ZH
)
@Retention(AnnotationRetention.SOURCE)
annotation class TypeMovies {
    companion object {
        const val API_TYPE_ALL_MOVIES: String = "Movies"
        const val API_TYPE_CHINESE_MOVIES: String = "ChineseMovies"
        const val API_TYPE_EURAMERICAN_MOVIES: String = "EuramericanMovies"
        const val API_TYPE_JANPAN_AND_KOREA_MOVIES: String = "JanpanAndKoreaMovies"
    }
}